### ISSUE TITLE
HDRP Auto-setup 1.4.9 crashes Unity 2022.3.0f1 when importing certain characters with Cloth and Hair Physics

### ISSUE THREAD

**Issue by AshD** [soupday/cc_unity_tools_HDRP#84]
Repro Steps:
1. Create character in Character Creator 4 using the Headshot plugin.
2. Export the FBX->Clothed Character to Unity
3. Import the character. Build Materials with Everything selected. 
4. View character in preview scene.
5. Run the scene in the editor. Editor crashes.
6. Import character again. Build Materials without Cloth and Hair Physics.
7. Run the scene in the editor. Editor works.

This happens with only certain characters. 
If you cannot repro this, I can send you a character that crashes.

Crash Log:

Loaded scene 'Temp/__Backupscenes/0.backup'
	Deserialize:            29.602 ms
	Integration:            222.794 ms
	Integration of assets:  0.010 ms
	Thread Wait Time:       -0.001 ms
	Total Operation Time:   252.405 ms

=================================================================
	Native Crash Reporting
=================================================================
Got a UNKNOWN while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Managed Stacktrace:
=================================================================
=================================================================
Received signal SIGSEGV
Obtained 18 stack frames
0x00007ff6e9d3ebf8 (Unity) NvClothCookFabricFromMesh
0x00007ff6e8371ac8 (Unity) ClothScene::TryBakeCloth
0x00007ff6e836e3ea (Unity) ClothScene::ProcessComponentUpdates
0x00007ff6e836d4c9 (Unity) ClothScene::PrepareScene
0x00007ff6e836cebe (Unity) ClothManager::LateUpdate
0x00007ff6e836c4ab (Unity) `ClothManager::Initialize'::`2'::PostLateUpdatePhysicsSkinnedClothBeginUpdateRegistrator::Forward
0x00007ff6e7a93f4a (Unity) ExecutePlayerLoop
0x00007ff6e7a940d6 (Unity) ExecutePlayerLoop
0x00007ff6e7a9a925 (Unity) PlayerLoop
0x00007ff6e8a5474f (Unity) PlayerLoopController::InternalUpdateScene
0x00007ff6e8a4bf53 (Unity) PlayerLoopController::EnterPlayMode
0x00007ff6e8a5c2a6 (Unity) PlayerLoopController::SetIsPlaying
0x00007ff6e8a5fa90 (Unity) Application::TickTimer
0x00007ff6e8eec56a (Unity) MainMessageLoop
0x00007ff6e8ef1310 (Unity) WinMain
0x00007ff6ea2d5d2e (Unity) __scrt_common_main_seh
0x00007ffaccb626ad (KERNEL32) BaseThreadInitThunk
0x00007ffacddaa9f8 (ntdll) RtlUserThreadStart

**Comment by soupday** [comment]
Couldn't reproduce it on my test characters.

Does it still crash in 2022.3.1f1?

Which hair and cloth meshes are you using that cause the crash?

**Comment by AshD** [comment]
Yes. It still crashes in 2023.3.1f1.

The hair is 'Chic bun medium.hairstyle' from the Stylings pack with the bangs removed. 
https://marketplace.reallusion.com/cc4-styling-pack

Clothes are 'Turtleneck sweater.cccloth' from Stylings pack and pants are 'Pants A5.cccloth' from the 'Outfits for walking, office and party Bundle. 
https://www.reallusion.com/ContentStore/Pack/outfits-for-walking-office-and-party-bundle-r1

Thanks,
Ash

**Comment by soupday** [comment]
Yes I could reproduce it. It was the one the hair meshes which wasn't generating any physics coefficients and crashed the physX native code. Managed to fix it and a few other problems, so the cloth physics should simulate and perform better.

Should be releasing the next version soon.
